### PR TITLE
Create github-fetcher to download and save github

### PR DESCRIPTION
**Pull request by apelisse**
This new program downloads and saves github database for a specific
project into a local database (MySql). The goal is to be able to query
this database without worrying about rate-limiting, and using a SQL
query language.

It currently downloads 2 types of resources:
- Issues, including pull-requests and including labels
- IssueEvents, that is all events associated to Issues

It is intended to run in a container and fetch/sleep/loop.

It is missing tests right now but it's big enough already.

### CODE REVIEW

File: github-fetcher/issue-events.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+func wasIdFound(events []github.IssueEvent, id int) bool {
+	for _, event := range events {
+		if *event.ID == id {
+			return true
+		}
+	}
+	return false
+}
+
+// Find most recent and oldest event in database
```
Comment (apelisse) [inline github-fetcher/issue-events.go]: Comment is not up-to-date

File: github-fetcher/github-fetcher.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	return nil
+}
+
+func main() {
+	config := &fetcherConfig{}
+	root := &cobra.Command{
+		Use:   filepath.Base(os.Args[0]),
+		Short: "Fetches entire github database: Pull-requests, issues, comments, activity",
```
Comment (apelisse) [inline github-fetcher/github-fetcher.go]: description is not up-to-date

**Comment by ingvagabund** [comment]
@apelisse could you break your code into more commits so it is easier to review? Something like:

- db implementation (with its part of vendored deps)
- fetcher implementation (with its part of vendored deps)
- client implementation (with its part of vendored deps)
- container definition (Dockerfile)
- yaml files

proposal/documentation with use cases and how to use it would be appreciated.

**Comment by apelisse** [comment]
@ingvagabund Sure!

**Comment by apelisse** [comment]
@ingvagabund, I've updated/splitted/added tests/added readme

Thanks for your review

**Comment by lavalamp** [comment]
First comment: please reuse/extend https://github.com/kubernetes/contrib/tree/master/mungegithub/github, let's not have two github access libraries...

More later.

**Comment by lavalamp** [comment]
Sorry-- I really don't have time to review this right now or soon :(

@fejta

**Comment by fejta** [comment]
I agree with lavalamp about using and improving our existing tools to access github instead of creating new ones.

Also please vendor at the repo level, and decouple any vendoring commits from everything else. Please assign back to me when this is complete.

**Comment by fejta** [comment]
Please move the README and initial commit into its own PR (in the test-infra repo), combine the README and initial commit together, move the vendoring into its own commit and send me that to review.

Thanks!

**Comment by apelisse** [comment]
Moving this pull-request to kubernetes/test-infra.

**Pull request closed without merging**